AdHoc (cert-creation) Enable recreation of certificates

toxix opened the pull request:
It turns out that old certificates are not renewed or updated with new
configurations (for example additional or changed domains) while
delployments.

It turns out that removing the certifificate in the path
/etc/ssl/certs/XXXX.lets_encrypt.crt
is not touched even when it should be changed. Removing this file is
getting the renewing process into success. This indicates that the
let's encrypt component is not overwriting existing files.

This now fixed with writing the new certificate into an temporary
directory and copy it over to the cert dir later with ansible.

Commit:
AdHoc (cert-creation) Enable recreation of certificates

It turns out that old certificates are not renewed or updated with new
configurations (for example additional or changed domains) while
delployments.

It turns out that removing the certifificate in the path
/etc/ssl/certs/XXXX.lets_encrypt.crt
is not touched even when it should be changed. Removing this file is
getting the renewing process into success. This indicates that the
let's encrypt component is not overwriting existing files.

This now fixed with writing the new certificate into an temporary
directory and copy it over to the cert dir later with ansible.

aboritskiy commented on tasks/lets_encrypt.yml:
```diff
@@ -63,7 +63,7 @@
     agreement: "{{ lets_encrypt_agreement }}"
     csr: "/etc/ssl/requests/{{ lets_encrypt_resource_name }}.csr"
-    dest: "/etc/ssl/certs/{{ lets_encrypt_resource_name }}.crt"
+    dest: "{{ build_dir.path }}/01-{{ lets_encrypt_resource_name }}.crt"
```
please use the convention that variable names should start with the role name

toxix replied:
Changed the variable name to follow the convention.

aboritskiy commented on tasks/lets_encrypt.yml:
```diff
@@ -63,7 +63,7 @@
     agreement: "{{ lets_encrypt_agreement }}"
     csr: "/etc/ssl/requests/{{ lets_encrypt_resource_name }}.csr"
-    dest: "/etc/ssl/certs/{{ lets_encrypt_resource_name }}.crt"
+    dest: "{{ build_dir.path }}/01-{{ lets_encrypt_resource_name }}.crt"
```
please provide a default value for the variable under `defaults/main.yml`

toxix replied:
Not required as this variable is filled during the runtime from the script and not ment to be changed from the outside of the script.

For refernce have look at line 11:
https://github.com/sitewards/ansible-role-lets-encrypt/pull/11/files/5571c3f2e5148d72fea54a2693a0a7d1472aad1e..ef7871590df0e2464af73d04b433f122ef6865cb#diff-7a113781da4adef4962744aaf7fb9a56L11

Commit:
AdHoc (cert-creation) follow naming variabe convention

It it commen standard to prefix ansible varialbe names with the
module name. This is renaming the variable to follow this
convention.

Commit:
Adhoc (cert creation) Do not recreate cert untill required

Let's encrypt api has an usage limit. To prevent us from running into we should
not send cert requests to them, that are not required because expiery is far in
the future (default 30 days or more).

This change is also testing if there are changes to the hosts that are validated
into this certificate. It will recreate the certificate, if there are changes
to the certificate subject.
